test/Frontend/module-alias-scan-deps.swift

```diff
@@ -0,0 +1,67 @@
+/// Test -scan-dependencies module aliasing.
+///
+/// Module 'Lib' imports module 'XLogging' via module aliasing and with various import attributes.
+/// Module 'User' imports 'Lib'.
+
+// RUN: %empty-directory(%t)
+// RUN: %{python} %utils/split_file.py -o %t %s
+
+/// Create AppleLogging.swiftmodule by aliasing XLogging
+// RUN: %target-swift-frontend -module-name AppleLogging -module-alias XLogging=AppleLogging %t/FileLogging.swift -emit-module -emit-module-path %t/AppleLogging.swiftmodule
+// RUN: test -f %t/AppleLogging.swiftmodule
+
+/// Scaned dependencies with implicit builds should contain real name AppleLogging
+// RUN: %target-swift-frontend -scan-dependencies  %t/FileLib.swift -module-alias XLogging=AppleLogging -I %t > %t/scandump.output
+// RUN: %FileCheck %s -check-prefix=CHECK-REAL-NAME -input-file  %t/scandump.output
+// CHECK-REAL-NAME-NOT: "swiftPrebuiltExternal": "XLogging"
+// CHECK-REAL-NAME-NOT: "compiledModulePath":{{.*}}XLogging.swiftmodule",
+// CHECK-REAL-NAME: "swiftPrebuiltExternal": "AppleLogging"
+// CHECK-REAL-NAME: "compiledModulePath":{{.*}}AppleLogging.swiftmodule",
+
+/// Scaned dependencies with explicit builds should contain real name AppleLogging
+// RUN: %target-swift-frontend -scan-dependencies -disable-implicit-swift-modules %t/FileLib.swift -module-alias XLogging=AppleLogging -I %t > %t/scandumpExplicit.output
```

Comment:
`-disable-implicit-swift-modules` is not necessary here, it should not have an effect during scanning.